### DISCUSSION TITLE
Get Document Type using Nodejs

### DISCUSSION THREAD

**Discussion by felipe-muner** [regulaforensics/DocumentReader-web-js-client#46] [Q&A] [answered]
Hello folks, thank you so much in advance for your effort...

I have been processing documents and everything is going great. I like using Regula and currently I'm using the following code:

![![![image](https://user-images.githubusercontent.com/2159694/130407202-1efe47be-d5c0-4690-9734-2b5d54a18296.png)](https://user-images.githubusercontent.com/2159694/130407165-8762838b-2456-493e-b906-8620c86bceb8.png)](https://user-images.githubusercontent.com/2159694/130406878-6dde23eb-5e24-4741-bfc1-d37787edc81e.png)

```
import { Context } from '../Context'
const configurationProviderService = require('../services/configurationProviderService')

class RegulaForensicsDocumentReaderService {
  public async processDocument({ documentImages }: { documentImages: string[] }, context: Context): Promise<Response> {
    const processed = await this.regulaForensicsDocumentReaderService(context).process({ images: documentImages })
    console.log(processed)
    return processed
  }
  ```
  
I changed as the answer in this previous discussion number 29. I'm not sure if I need to do something else...

![image](https://user-images.githubusercontent.com/2159694/130407211-ef7a388e-ef12-4c4f-b47d-9d01f240374a.png)

And I have something as the response below:

![image](https://user-images.githubusercontent.com/2159694/130407297-148bb338-3dce-46f2-8c9d-46543a8cac91.png)

I even stop the service and restart it, I have not idea about what I need to change now...

What am I missing?

I've tried passing the processParam as well.

![image](https://user-images.githubusercontent.com/2159694/130407338-487a384c-f112-465e-9891-48c2cca05fc7.png)

As we have it for C# we can have it here as well for JS developer. We know there will be more people asking the same :)...

Cheers =)

**Comment by felipe-muner** [comment, marked as answer]
Actually I guess I've found it.

![image](https://user-images.githubusercontent.com/2159694/130413354-00b5e0d3-43f2-48b9-a8aa-257522695795.png)

**Reply by hleb-albau** [reply]
Yes, you just overtook me by a few minutes :)

```
  // ChosenDocumentTypeResult type
  const docTypeResult = response.lowLvlResponse.resultByType(Result.DOCUMENT_TYPE)
  const docType = docTypeResult.OneCandidate.FDSIDList.dType
```

[Here is](https://github.com/regulaforensics/DocumentReader-web-openapi/blob/master/e-document-type.yml) an enum with values description.

**Reply by felipe-muner** [reply]
Thank you for your answer, I understood an extra step, we can user the function `.resultByType(Result.DOCUMENT_TYPE)`

**Comment by felipe-muner** [comment]
I have another question and maybe you can help me... Even though I'm trying to search it...

There are many values for passports, as there many values for driver license...

Is it possible to get all values that are from DRIVING_LICENSE group or PASSPORT group... and so on... I can do by myself but maybe it will not be 100%...

all values below are related to DRIVING_LICENSE....

    dtDrivingLicense = 49,
    dtDrivingLicenseInstructionalPermit = 50,
    dtDrivingLicenseInstructionalPermitUnder18 = 51,
    dtDrivingLicenseInstructionalPermitUnder21 = 52,
    dtDrivingLicenseLearnersPermit = 53,
    dtDrivingLicenseLearnersPermitUnder18 = 54,
    dtDrivingLicenseLearnersPermitUnder21 = 55,
    dtDrivingLicenseNovice = 56,
    dtDrivingLicenseNoviceUnder18 = 57,
    dtDrivingLicenseNoviceUnder21 = 58,
    dtDrivingLicenseRegisteredOffender = 59,
    dtDrivingLicenseRestrictedUnder18 = 60,
    dtDrivingLicenseRestrictedUnder21 = 61,
    dtDrivingLicenseTemporaryVisitor = 62,
    dtDrivingLicenseTemporaryVisitorUnder18 = 63,
    dtDrivingLicenseTemporaryVisitorUnder21 = 64,
    dtDrivingLicenseUnder18 = 65,
    dtDrivingLicenseUnder21 = 66,
    dtEmploymentDrivingPermit = 67,
    dtEnhancedChauffeurLicense = 68,
    dtEnhancedChauffeurLicenseUnder18 = 69,
    dtEnhancedChauffeurLicenseUnder21 = 70,
    dtEnhancedCommercialDrivingLicense = 71,
    dtEnhancedDrivingLicense = 72,
    dtEnhancedDrivingLicenseUnder18 = 73,
    dtEnhancedDrivingLicenseUnder21 = 74,
    //and more....


    Passport we have something like it:
    dtPassport
    dtDiplomaticPassport
    dtServicePassport
    dtEmergencyPassport
    dtAliensPassport
    dtPassportCard
    dtTemporaryPassport
    dtPassportStateless
    dtPassportChild
    dtPassportConsular
    dtPassportDiplomaticService
    dtPassportOfficial
    dtPassportProvisional
    dtPassportSpecial
    dtVehiclePassport
    dtPassportLimitedValidity
    // and maybe more... 

And I would like to have for national ID as well...

**Comment by esemenov1** [comment]
Hello Felipe!

Please find the .xlsx file with the necessary information attached. Feel free to contact me at support@regulaforensics.com
Eugene S.
[result.xlsx](https://github.com/regulaforensics/DocumentReader-web-js-client/files/7040765/result.xlsx)
